dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/VisualizationController.java

```diff
@@ -141,6 +150,30 @@ public void postProcessResponseEntity( Visualization visualization, WebOptions o
                     period.setName( i18nFormat.formatPeriod( period ) );
                 }
             }
+
+            List<BaseDimensionalItemObject> dimensionalItemObjectList = visualization.getDataDimensionItems()
```

Comment:
We could extract the new code into a new method, ie. `setDataDimensionItems (visualization)`.

Reply:
done